internal/storage/storageutil/custom_retry.go

```diff
@@ -0,0 +1,78 @@
+package storageutil
+
+import (
+	"io"
+	"net"
+	"net/url"
+
+	"google.golang.org/api/googleapi"
+)
+
+func ShouldRetry(err error) (b bool) {
+	// HTTP 50x errors.
+	if typed, ok := err.(*googleapi.Error); ok {
+		if typed.Code >= 500 && typed.Code < 600 {
```

Comment:
did you explicitly validate each error from the go-client library and that its getting retried. It might happen that we are expecting googleapi.error in case 5xx and go-client library is returned something else.